### PR TITLE
Remove autofollowing hiring tag

### PR DESCRIPTION
**Pull request by citizen428**
## What type of PR is this? (check all applicable)

- [X] Refactor

## Description

This PR removes the automatic following of the hiring tag after ticking the "Looking for work" checkbox on a user's profile. This was primarily done for two reasons:

1. It's very DEV specific and dates back to a time where DEV was still exploring other possible futures than Forem.
2. The hiring fields are pretty complex, as they rely on the interaction of two fields (are you looking at all and if yes, is it public?) which doesn't generalize well.

Because of 2. I vote for completely removing the "looking for work publicly" field from all Forems via a data update script once the generalized profile page has been merged (see #11485) as it will no longer serve a purpose. We might even do that as part of the other PR.

I'll leave this PR as draft for now, as we still have some product-related discussion going on in https://github.com/forem/InternalProjectPlanning/issues/366 as to how to notify users who currently have these flags set about the upcoming change.

## Related Tickets & Documents

Closes https://github.com/forem/InternalProjectPlanning/issues/366

## QA Instructions, Screenshots, Recordings

0. Ensure your user doesn't follow the `hiring` tag.
1. Go to `/settings/profile`
2. Check the "Looking for work" checkbox and save your profile:
    Screen Shot 2020-12-03 at 12 22 15
3. Your user should still not follow the `hiring` tag.

## Added tests?

- [X] No, and this is why: I'm removing a feature, so I'm also removing the corresponding spec.

## Added to documentation?

- [X] No documentation needed

## [optional] What gif best describes this PR or how it makes you feel?

image

### CODE REVIEW

**Review by rhymes — APPROVED** [review]
Bye bye :D

File: spec/system/user_selects_looking_for_work_spec.rb [hunk 1]
```diff
@@ -1,21 +0,0 @@
...
-
-    sign_in(user)
-    visit "/settings"
-    page.check("profile[looking_for_work]")
-    sidekiq_perform_enqueued_jobs { click_button("Save") }
-
-    expect(page).to have_text("Your profile has been updated")
-    expect(user.follows.count).to eq(1)
-  end
-end
```
Comment (vaidehijoshi) [inline spec/system/user_selects_looking_for_work_spec.rb:21]: 🔥 ✂️

File: app/services/profiles/update.rb [hunk 2]
```diff
@@ -23,10 +23,6 @@ def call
       if update_successful?
         @success = true
         @user.touch(:profile_updated_at)
-        # TODO: @citizen428 Preserving a DEV specific feature for now, we should
-        # probably remove this sooner than later as it may not make much sense
-        # for other communities.
-        follow_hiring_tag if SiteConfig.dev_to?
```
Comment (vaidehijoshi) [inline app/services/profiles/update.rb:29]: I dream of the day we can remove `.dev_to?` altogether! We are getting so close 😍

**Review by vaidehijoshi — APPROVED** [review]
hooray!!

File: app/services/profiles/update.rb (hunk 2 above)
Comment (citizen428) [reply app/services/profiles/update.rb:29]: Yes @vaidehijoshi, definitely a goal I'm working towards.

**Pull request merged**